src/Scientist/Internals/Experiment.cs

```diff
@@ -68,19 +72,36 @@ public void Try(string name, Func<T> candidate)
         {
             if (_candidates.ContainsKey(name))
             {
-                throw new InvalidOperationException($"You already have a candidate named {name}. Provide a different name for this test.");
+                throw new InvalidOperationException(
+                    $"You already have a candidate named {name}. Provide a different name for this test.");
             }
             _candidates.Add(name, () => Task.FromResult(candidate()));
         }
 
-        public void Ignore(Func<T, T, bool> block) => 
+        public void Ignore(Func<T, T, bool> block) =>
             _ignores.Add((con, can) => Task.FromResult(block(con, can)));
 
-        public void Ignore(Func<T, T, Task<bool>> block) => 
+        public void Ignore(Func<T, T, Task<bool>> block) =>
             _ignores.Add(block);
 
+        public void AddContext(string key, dynamic data)
```

Comment:
I didn't see a need to catch/rethrow/throw-our-own here and let it bubble the `ArgumentException` up if a duplicate key is added.